pandas/core/arrays/arrow/dtype.py

```diff
@@ -132,6 +133,14 @@ def type(self):
             return time
         elif pa.types.is_decimal(pa_type):
             return Decimal
+        elif pa.types.is_dictionary(pa_type):
+            # TODO: Potentially change this & CategoricalDtype.type to
+            #  something more representative of the scalar
+            return CategoricalDtypeType
+        elif pa.types.is_list(pa_type) or pa.types.is_large_list(pa_type):
```

Comment:
side note: think with this supported we can get rid of JSONArray?

Reply:
Yeah I think we can get rid of the date and decimal arrays now since they have corresponding arrow types tested and  json and list arrays once we add tests for those arrow types